### ISSUE TITLE
Variables visibility between blocks depend on block order

### ISSUE THREAD

**Issue by milo** [nette/latte#178]
Version: master (f7b0fa8717de7d43f06d459b8b2d1b20d4c11f9b)

Following code emits notice: `PHP Notice: Undefined variable: x`

```php
<?php
# composer require latte/latte dev-master
require __DIR__ . '/vendor/autoload.php';

$latte = '
{define a}
	{var x => "X"}
	{include #b}
{/}

{define b}
	{$x}
{/}

{include a}
';

$engine = new Latte\Engine;
$engine->setLoader(new Latte\Loaders\StringLoader);
echo $engine->compile($latte);
$engine->render($latte);
```
To make it work, I have to pass `$x` explicitly to block. Or, If I change blocks order, it works:
```php
$latte = '
{define b}
	{$x}
{/}

{define a}
	{var x => "X"}
	{include #b}
{/}

{include a}
';
```

The only difference between compiled template is in method `blockA()`:
```php
# Non working
function blockA($_args)
{
        extract($_args);
        $x = "X";
        $this->renderBlock('b', $this->params, 'html');
}

# Working
function blockA($_args)
{
        extract($_args);
        $x = "X";
        $this->renderBlock('b', get_defined_vars(), 'html');
}
```

I'm not sure it is bug or by desing.

**Comment by matej21** [comment]
it's related to #163

**Comment by milo** [comment]
Thank you!

**Comment by solcik** [comment]
@dg Hello, this template was ok till v2.9.0, but with v2.9.1 is throwing warning, that variable `$T` is not defined.

When I skip the warning, the translator is working just fine.
This use-case is also mentioned in the docs: https://latte.nette.org/en/template-inheritance

> Data outside of a blocks in a child template is executed before the layout template is rendered, thus you can use it to define variables like {var $foo = bar} and propagate data to the whole inheritance chain:

```latte
{templateType HomepageTemplate}
{var $T = 'app.' . HomepagePresenter::class}

{block h1}
    {_"$T.heading.h1"}
{/block}

{block content}
    <div></div>
{/block}
```

**Comment by filipmelik** [comment]
@solcik I had very similar code to yours and this stopped working in v2.9.1:

Template code:
```
{var $sanitizedEmployeeDisplayName = "TEST TEST"}

{block title}
    {_messages.pageTitle.job-detail} {$sanitizedEmployeeDisplayName}
{/block}

{block content}

    <div class="d-flex justify-content-between flex-wrap flex-md-nowrap align-items-center pt-3 pb-2 mb-3 border-bottom">
        <h1 class="h2">
           {$sanitizedEmployeeDisplayName} - {_messages.labour.job.detail.title}
        </h1>
    </div>
{/block}
```

@layout.latte (template base) - simplified example:
```
<!doctype html>
<html lang="en" class="js">
<head>
    <title>{block title|stripHtml}MEOW{/block}</title>
</head>

<body>
    {include #content}
</body>
</html>
```

When i changed @layout.latte html body tag content to from `{include #content}` to `{block content}` it started to work again. I guess it now "replaces" the body tag including its context from the "child" template and thus it works.

**Comment by dg** [comment]
@solcik @filipmelik fixed in 2.9-dev
